go.mod

```diff
@@ -87,7 +87,7 @@ require (
 	go.uber.org/atomic v1.10.0
 	go.uber.org/multierr v1.11.0
 	go.uber.org/zap v1.24.0
-	go.viam.com/api v0.1.311
+	go.viam.com/api v0.1.315
```

Comment:
Looking at the failures on the CI for this PR, it looks like some `api` version between 311 and 315 included a change that deprecated some method the RDK code is currently using. Could you look around to see if there's an existing `rdk` PR to fix the usage of the deprecated method?

Reply:
Do deprecated proto methods actually need to be removed? I don't think we're quite ready to do that yet.

Reply:
ya, I didn't realize but we need to add corresponding `//nolint:staticcheck` to all the places that error - the rdk team did that for when they deprecated sensors service

Reply:
@maximpertsov or @puranamp can you guys update?

Reply:
merged https://github.com/viamrobotics/rdk/pull/4131